backend/app/services/embedding_service/service.py

```
@@ -1,225 +1,293 @@
+        if self._tokenizer is None:
+            with self._tokenizer_lock:
+                if self._tokenizer is None:
+                    self._tokenizer = tiktoken.get_encoding("cl100k_base")
+        return self._tokenizer
+
+    def _count_tokens(self, text: str) -> int:
+        if self.tokenizer:
+            try:
+                return len(self.tokenizer.encode(text))
+            except Exception:
+                pass
```

Ruff (0.14.10)

`try`-`except`-`pass` detected, consider logging the exception

(S110)